diff -urN miniupnpd/linux-org/iptables_init.sh miniupnpd/linux/iptables_init.sh
--- miniupnpd/linux-org/iptables_init.sh        2006-12-05 23:17:16.000000000 -0600
+++ miniupnpd/linux/iptables_init.sh    2006-12-05 23:19:50.000000000 -0600
@@ -3,7 +3,10 @@
 IPTABLES=iptables

 #change this parameters :
-EXTIF=eth0
+include /lib/network
+scan_interfaces
+config_get ifname wan ifname
+EXTIF=$ifname
 EXTIP="`LC_ALL=C /sbin/ifconfig $EXTIF | grep 'inet addr' | awk '{print $2}' | sed -e 's/.*://'`"
 echo "External IP = $EXTIP"

 
diff -urN miniupnpd/linux-org/iptables_removeall.sh miniupnpd/linux/iptables_removeall.sh
--- miniupnpd/linux-org/iptables_removeall.sh   2006-12-05 23:17:16.000000000 -0600
+++ miniupnpd/linux/iptables_removeall.sh       2006-12-05 23:21:03.000000000 -0600
@@ -3,7 +3,10 @@
 IPTABLES=iptables

 #change this parameters :
-EXTIF=eth0
+include /lib/network
+scan_interfaces
+config_get ifname wan ifname
+EXTIF=$ifname
 EXTIP="`LC_ALL=C /sbin/ifconfig $EXTIF | grep 'inet addr' | awk '{print $2}' | sed -e 's/.*://'`"

 #removing the MINIUPNPD chain for nat
