diff -urN file-4.21.old/magic/Magdir/apple file-4.21.new/magic/Magdir/apple
--- file-4.21.old/magic/Magdir/apple	2006-03-02 23:10:26.000000000 +0100
+++ file-4.21.new/magic/Magdir/apple	2007-10-29 14:37:46.000000000 +0100
@@ -10,6 +10,23 @@
 0	belong		0x00051600		AppleSingle encoded Macintosh file
 0	belong		0x00051607		AppleDouble encoded Macintosh file
 
+# Apple Emulator 2IMG format
+#
+0	string		2IMG		Apple ][ 2IMG Disk Image
+>4	string		XGS!		\b, XGS
+>4	string		CTKG		\b, Catakig
+>4	string		ShIm		\b, Sheppy's ImageMaker
+>4	string		WOOF		\b, Sweet 16
+>4	string		B2TR		\b, Bernie ][ the Rescue
+>4	string		!nfc		\b, ASIMOV2
+>4	string		x		\b, Unknown Format
+>0xc	byte		00		\b, DOS 3.3 sector order
+>>0x10	byte		00		\b, Volume 254
+>>0x10	byte&0x7f	x		\b, Volume %u
+>0xc	byte		01		\b, ProDOS sector order
+>>0x14	short		x		\b, %u Blocks
+>0xc	byte		02		\b, NIB data
+
 # magic for Newton PDA package formats
 # from Ruda Moura <ruda@helllabs.org>
 0	string	package0	Newton package, NOS 1.x,
diff -urN file-4.21.old/magic/Magdir/images file-4.21.new/magic/Magdir/images
--- file-4.21.old/magic/Magdir/images	2007-05-03 17:11:33.000000000 +0200
+++ file-4.21.new/magic/Magdir/images	2007-10-29 14:37:46.000000000 +0100
@@ -525,6 +525,12 @@
 0	belong	0x0e031301	Hierarchical Data Format (version 4) data
 0	string	\211HDF\r\n\032	Hierarchical Data Format (version 5) data
 
+# The boot loaders syslinux and isolinux use a RLE based image format
+# called SLL16 to store splash screens.
+0	lelong		0x1413f33d	Syslinux SLL16 image data,
+>4	leshort		>0		%hd x
+>6	leshort		>0		%hd
+
 # From: Tobias Burnus <burnus@net-b.de>
 # Xara (for a while: Corel Xara) is a graphic package, see
 # http://www.xara.com/ for Windows and as GPL application for
diff -urN file-4.21.old/magic/Magdir/revision file-4.21.new/magic/Magdir/revision
--- file-4.21.old/magic/Magdir/revision	2007-04-09 18:33:07.000000000 +0200
+++ file-4.21.new/magic/Magdir/revision	2007-10-29 14:37:46.000000000 +0100
@@ -4,6 +4,11 @@
 # From Hendrik Scholz <hendrik@scholz.net>
 0	string	/1\ :pserver:	cvs password text file
 
+# Subversion (SVN) dumps
+# Uwe Zeisberger <zeisberg@informatik.uni-freiburg.de>
+0       string  SVN-fs-dump-format-version:     Subversion dumpfile
+>28     string  >\0                             (version: %s)
+
 # Conary changesets
 # From: Jonathan Smith <smithj@rpath.com>
 0	belong	0xea3f81bb	Conary changeset data
diff -urN file-4.21.old/magic/magic.mime file-4.21.new/magic/magic.mime
--- file-4.21.old/magic/magic.mime	2007-04-03 23:11:32.000000000 +0200
+++ file-4.21.new/magic/magic.mime	2007-10-29 14:37:46.000000000 +0100
@@ -196,7 +196,7 @@
 # modified by Joerg Jenderek
 # GRR the original test are too common for many DOS files
 # so test 1 <= kbits nibble <= E
-0       beshort		&0xffe0		
+0       beshort&0xfffe  =0xfffa         audio/mpeg
 >2	ubyte&0xF0	>0x0F		
 >>2	ubyte&0xF0	<0xE1		audio/mpeg
 #MP3 with ID3 tag
@@ -402,16 +402,14 @@
 #
 # from Daniel Quinlan <quinlan@yggdrasil.com>
 #
-0	string		\<HEAD	text/html
-0	string		\<head	text/html
-0	string		\<TITLE	text/html
-0	string		\<title	text/html
-0       string          \<html	text/html
-0       string          \<HTML	text/html
+0	string/cB	\<!DOCTYPE\ html	text/html
+0	string/cb	\<head	text/html
+0	string/cb	\<title	text/html
+0       string/bc	\<html	text/html
 0	string		\<!--	text/html
-0	string		\<h1	text/html
-0	string		\<H1	text/html
-0	string/c	\<!doctype\ html	text/html
+0	string/c	\<h1	text/html
+
+0	string		\<?xml			text/xml
 
 #------------------------------------------------------------------------------
 # images:  file(1) magic for image formats (see also "c-lang" for XPM bitmaps)
@@ -474,7 +472,7 @@
 0	beshort		0xffd8		image/jpeg
 
 # PC bitmaps (OS/2, Windoze BMP files)  (Greg Roelofs, newt@uchicago.edu)
-0	string		BM		image/bmp
+0	string		BM		image/x-ms-bmp
 #>14	byte		12		(OS/2 1.x format)
 #>14	byte		64		(OS/2 2.x format)
 #>14	byte		40		(Windows 3.x format)
@@ -939,6 +937,12 @@
 0	string	fLaC		audio/x-flac
 0	string	CWS		application/x-shockwave-flash
 
+
+# Gnumeric spreadsheet
+# This entry is only semi-helpful, as Gnumeric compresses its files, so
+# they will ordinarily reported as "compressed", but at least -z helps
+39      string          =<gmr:Workbook           application/x-gnumeric
+
 # Hangul Document Files:
 # Reversed-engineered HWP magic numbers
 # From: Won-Kyu Park <wkpark@kldp.org>
diff -urN file-4.21.old/src/fsmagic.c file-4.21.new/src/fsmagic.c
--- file-4.21.old/src/fsmagic.c	2007-01-12 18:40:53.000000000 +0100
+++ file-4.21.new/src/fsmagic.c	2007-10-29 14:37:46.000000000 +0100
@@ -92,7 +92,8 @@
 		if (file_printf(ms, "cannot open `%s' (%s)",
 		    fn, strerror(errno)) == -1)
 			return -1;
-		return 1;
+		ms->haderr++;
+		return -1;
 	}
 
 	if ((ms->flags & MAGIC_MIME) != 0) {
