diff -urN xyssl-0.8.old/library/Makefile xyssl-0.8.new/library/Makefile
--- xyssl-0.8.old/library/Makefile	2007-10-20 00:00:00.000000000 +0200
+++ xyssl-0.8.new/library/Makefile	2007-10-24 14:56:11.000000000 +0200
@@ -4,6 +4,8 @@
 CFLAGS	= -I../include
 OFLAGS	= -O
 
+AR	?= ar
+
 # MicroBlaze specific options:
 # CFLAGS += -mno-xl-soft-mul -mxl-barrel-shift
 
@@ -11,7 +13,7 @@
 # CFLAGS += -D_BSD_EXTENSION
 
 # To compile as a shared library:
-# CFLAGS += -fPIC
+CFLAGS += -fPIC
 
 DLEXT=so
 # OSX shared library extension:
@@ -28,7 +30,7 @@
 
 .SILENT:
 
-all: static
+all: static shared
 
 static: libxyssl.a
 
@@ -36,7 +38,7 @@
 
 libxyssl.a: $(OBJS)
 	echo "  AR    $@"
-	ar r $@ $(OBJS)
+	$(AR) r $@ $(OBJS)
 
 libxyssl.so: libxyssl.a
 	echo "  LD    $@"
