--- miniupnpd-1.0-RC8.old/linux/iptcrdr.c	2007-07-17 00:52:33.000000000 +0200
+++ miniupnpd-1.0-RC8.dev/linux/iptcrdr.c	2007-11-21 15:50:12.000000000 +0100
@@ -16,12 +16,12 @@
 #include <libiptc/libiptc.h>
 #include <iptables.h>
 
-#include <linux/version.h>
-#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,22)
+/*#include <linux/version.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,22)*/
 #include <linux/netfilter_ipv4/ip_nat.h>
-#else
+/*#else
 #include <linux/netfilter/nf_nat.h>
-#endif
+#endif*/
 
 #include "iptcrdr.h"
 
